zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -326,11 +333,26 @@ private[codec] object EncoderDecoder                   {
     }
 
     private def encodeContentType(inputs: Array[Any]): Headers = {
-      val _ = inputs // TODO: Support multiple content types
-      if (jsonEncoders.length == 0) Headers.empty
-      else if (jsonEncoders.length == 1) {
-        Headers(Header.ContentType(MediaType.application.json))
-      } else throw new IllegalStateException("A request on a REST endpoint should have at most one body")
+      if (isByteStream(flattened.content)) {
+        Headers(Header.ContentType(MediaType.application.`octet-stream`)) // TODO: customizable content type
```

Comment:
We should be able to attach a custom media type to the body codec somehow, not sure if it should be part of this PR or not

Reply:
I think it should be part of this PR. Basically, store an `Option[MediaType]` inside the codec, and add variants that allow it to be specified (overridden) when the content codec is specified.